app/controllers/admin/subjects_controller.rb

```diff
@@ -0,0 +1,38 @@
+class Admin::SubjectsController < Admin::ApplicationController
+  include Admin::ApplicationHelper
+
+  before_action :get_subject, only: [:destroy]
+  before_action :extract_params, only: [:index]
+
+  def index
+    @subjects = Subject.find_by_name(@search)
+                       .paginate(page: @page, per_page: @per_page)
+    flash.now[:danger] = t("controller.admin.controller.subject.index.no_result") if !@search.empty? && @subjects.count == 0
+  end
+
+  def destroy
+    @subject.destroy
+    if @subject.destroyed?
+      redirect_to admin_subjects_path,
+                  flash: {success: t("controller.admin.controller.subject.destroy.deleted")}
+    else
+      redirect_to admin_subjects_path,
+                  flash: {danger: t("controller.admin.controller.subject.destroy.failed")}
+    end
+  end
+
+  private
+  def get_subject
+    @subject = Subject.find_by(id: params[:id])
+    unless @subject.present?
+      redirect_to admin_subjects_path,
+                  flash: {danger: t("controller.admin.controller.subject.general.not_found")}
+    end
```

Comment:
từ line 27 đến line 30 sửa thành ->
return if @subject
flash[:warning] = "Subject not found"
redirect_to root_paht